govcd/vdcComputePolicy.go

```diff
@@ -0,0 +1,246 @@
+	if err != nil {
+		return nil, err
+	}
+
+	returnVdcComputePolicy := &VdcComputePolicy{
+		VdcComputePolicy: &types.VdcComputePolicy{},
+		client:           vdcComputePolicy.client,
+	}
+
+	err = vdcComputePolicy.client.OpenApiPutItem(minimumApiVersion, urlRef, nil, vdcComputePolicy.VdcComputePolicy, returnVdcComputePolicy.VdcComputePolicy)
+	if err != nil {
+		return nil, fmt.Errorf("error updating VDC compute policy: %s", err)
+	}
+
+	return returnVdcComputePolicy, nil
+}
+
+// Delete deletes VDC compute policy
+func (vdcComputePolicy *VdcComputePolicy) Delete() error {
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointVdcComputePolicies
+	minimumApiVersion, err := vdcComputePolicy.client.checkOpenApiEndpointCompatibility(endpoint)
+	if err != nil {
+		return err
+	}
+
+	if vdcComputePolicy.VdcComputePolicy.ID == "" {
+		return fmt.Errorf("cannot delete VDC compute policy without id")
+	}
+
+	urlRef, err := vdcComputePolicy.client.OpenApiBuildEndpoint(endpoint, vdcComputePolicy.VdcComputePolicy.ID)
+	if err != nil {
+		return err
+	}
+
+	err = vdcComputePolicy.client.OpenApiDeleteItem(minimumApiVersion, urlRef, nil)
+
+	if err != nil {
+		return fmt.Errorf("error deleting VDC compute policy: %s", err)
+	}
+
+	return nil
+}
+
+// GetAllAssignedVdcComputePolicies retrieves all VDC assigned compute policies using OpenAPI endpoint. Query parameters can be supplied to perform additional
+// filtering
+func (vdc *AdminVdc) GetAllAssignedVdcComputePolicies(queryParameters url.Values) ([]*VdcComputePolicy, error) {
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointVdcAssignedComputePolicies
+	minimumApiVersion, err := vdc.client.checkOpenApiEndpointCompatibility(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	urlRef, err := vdc.client.OpenApiBuildEndpoint(fmt.Sprintf(endpoint, vdc.AdminVdc.ID))
+	if err != nil {
+		return nil, err
+	}
+
+	responses := []*types.VdcComputePolicy{{}}
+
+	err = vdc.client.OpenApiGetAllItems(minimumApiVersion, urlRef, queryParameters, &responses)
+	if err != nil {
+		return nil, err
+	}
+
+	var wrappedVcdComputePolicies []*VdcComputePolicy
+	for _, response := range responses {
+		wrappedVcdComputePolicy := &VdcComputePolicy{
+			client:           vdc.client,
+			VdcComputePolicy: response,
+		}
+		wrappedVcdComputePolicies = append(wrappedVcdComputePolicies, wrappedVcdComputePolicy)
+	}
+
+	return wrappedVcdComputePolicies, nil
+}
+
+// SetAssignedComputePolicies assign(set) compute policies.
+func (vdc *AdminVdc) SetAssignedComputePolicies(computePolicyReferences types.VdcComputePolicyReferences) (*types.VdcComputePolicyReferences, error) {
+	util.Logger.Printf("[TRACE] Set Compute Policies started")
+
+	if !vdc.client.IsSysAdmin {
+		return nil, fmt.Errorf("functionality requires system administrator privileges")
+	}
+
+	adminVdcPolicyHREF, err := url.ParseRequestURI(vdc.AdminVdc.HREF)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing vdc url: %s", err)
```

Comment:
Consistency, in all log messages should be capital `VDC` and `URL`:
```suggestion
		return nil, fmt.Errorf("error parsing VDC URL: %s", err)
```

Reply:
changed